packages/core-parse/README.md

```diff
@@ -0,0 +1,259 @@
+  - `afterTransforms` processors are operations which extract information from the syntax tree to the global state.
+
+## Extension mechanism
+
+Everything is an extension!
+
+```javascript
+import { u } from 'unist-builder'
+import { toc } from 'mdast-util-toc'
+import { Processor, RoleProcessor, DirectiveProcessor } from '@unified-myst/core-parse'
+
+class RoleAbbr extends RoleProcessor {
+    run() {
+        const abbr = u('abbr', [])
+        abbr.children = this.nestedInlineParse(this.node.content)
+        return [abbr]
+    }
+}
+
+class DirectiveNote extends DirectiveProcessor {
+    static has_content = true
+    run() {
+        const note = u('note', [])
+        note.children = this.nestedParse(this.node.body)
+        return [note]
+    }
+}
+
+function addToc(ast, config) {
+    if (config.myExtension?.addtoc) {
+        const table = toc(ast)
+        ast.children.unshift(table.map)
+    }
+}
+
+myExtension = {
+  name: 'myExtension',
+  roles: { abbr: { processor: RoleAbbr } },
+  directives: { note: { processor: DirectiveNote } },
+  hooks: { afterRead: { addtoc: { priority: 100, processor: addToc } } },
+  config: { addtoc: { default: false, type: 'boolean' } },
+}
+parser = Processor().use(myExtension)
+parser.setConfig({myExtension: {addtoc: true}})
+result = parser.toAst('hallo')
+```
+
+## Configuration
+
+Each extension can supply its own configuration, as a "stub" for the [`properties`](https://json-schema.org/understanding-json-schema/reference/object.html#properties) key of a JSON schema.
+
+```javascript
+import { Processor } from '@unified-myst/core-parse'
+
+const processor = new Processor()
+processor.use({
+    name: 'core',
+    config: {
+        cname1: { default: '', type: 'string' },
+        cname2: { default: [], type: 'array' },
+    },
+})
+```
+
+From these configuration stubs, the full configuration schema is generated.
+
+```javascript
+console.log(JSON.stringify(processor.getConfigSchema(), null, 2))
+// {
+//   "type": "object",
+//   "properties": {
+//     "core": {
+//       "type": "object",
+//       "properties": {
+//         "cname1": {
+//           "default": "",
+//           "type": "string"
+//         },
+//         "cname2": {
+//           "default": [],
+//           "type": "array"
+//         }
+//       },
+//       "additionalProperties": false
+//     }
+//   },
+//   "additionalProperties": false
+// }
+```
+
+A configuration can be set using the [`setConfig`] method, then the [`getConfig`] method can be used to retrieve the full configuration, which merges this config with any defaults.
+
+```javascript
+processor.setConfig({ core: { cname1: 'test' } })
+console.log(JSON.stringify(processor.getConfig(), null, 2))
+// {
+//   "core": {
+//     "cname1": "test",
+//     "cname2": []
+//   }
+// }
+```
+
+## Design decisions
+
+The design is intended to quite closely mirror that of [docutils](https://docutils.sourceforge.io) and [Sphinx](https://www.sphinx-doc.org).
```

Comment:
Would love to see some mention that this is moving to support scientific writing, and that decisions will be made to change/improve underlying pieces to go in that direction.

Reply:
Yep makes sense 👍

Reply:
added a line